### PR TITLE
Don't redirect stdlib wchar functions to our test platform shims.

### LINKED ISSUE #22069: Build failure: OSX: error: cannot initialize a parameter of type 'LPWSTR' (aka 'char16_t *') with an lvalue of type 'const wchar_t *'
Opened on behalf of @RussKeldorph

Warnings: 4
Status Message: failed

Build : 3.0 - 20190118.02 (Product Build)
Failing configurations:
- OSX
  - PortableBuild-Checked-x64
  - PortableBuild-Debug-x64
  - PortableBuild-Release-x64
[Mission Control Build Info](https://mc.dot.net/#/product/netcore/30/source/official~2Fcoreclr~2Fmaster~2F/type/build~2Fproduct~2F/build/20190118.02/workItem/Orchestration/analysis/external/Link)

### PR DESCRIPTION
**Pull request by jkoritzinsky**
We were redirecting the `wchar_t`-related standard library functions to our own implementations in our tests. This was convenient at the time since it would allow everything to work cross-platform. However, this design can cause us issues cross-plat since `wchar_t != WCHAR` on non-Windows platforms (as seen in #22069).

Fixes #22069.

### CODE REVIEW

File: tests/src/Common/Platform/platformdefines.cpp [hunk 1]
```diff
@@ -584,7 +584,7 @@ int TP_wcsncpy_s(LPWSTR strDestination, size_t size1, LPCWSTR strSource, size_t
 
 int TP_wcsncpy_s(LPWSTR strDestination, size_t size1, LPCWSTR strSource)
 {
-    return wcsncpy_s(strDestination, size1, strSource, 0);
+    return TP_wcsncpy_s(strDestination, size1, strSource, 0);
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply tests/src/Common/Platform/platformdefines.cpp]: Good catch. Missed that in my find-replace.

[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply tests/src/Common/Platform/platformdefines.cpp]: Actually, this one calls into a different overload, so we're good. But there were some other ones that I missed.

[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply tests/src/Common/Platform/platformdefines.cpp]: Wait a second here. If the name of the function is `TP_wcsncpy_s` it is clearly trying to mimic `wcsncpy_s` which means it has an identical signature. There should be no overloads for these kind of functions unless the standard says there are.

Alternatively we can create a new function that doesn't have such an implied meaning.

[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply tests/src/Common/Platform/platformdefines.cpp]: Looks like that overload is actually never used so I've removed it.

**Comment by jkoritzinsky** [comment]
@dotnet-bot test this please

**Comment by jkoritzinsky** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by jkoritzinsky** [comment]
The one CI failure here is a known failure with the new CI system, so I'm going to merge this in.

**Pull request merged**